app/models/TraderProfile.scala

```diff
@@ -80,4 +81,27 @@ object TraderProfile {
         }
       case Left(errors) => Left(errors)
     }
+
+  def validateHasNirms(
+    answers: UserAnswers
+  ): EitherNec[ValidationError, Boolean] =
+    answers.getPageValue(HasNirmsUpdatePage) match {
+      case Right(true)  => Left(NonEmptyChain.one(UnexpectedPage(HasNirmsUpdatePage)))
+      case Right(false) =>
+        answers.getPageValue(TraderProfileQuery) match {
+          case Right(userProfile) =>
+            if (userProfile.nirmsNumber.isDefined) {
+              answers.getPageValue(RemoveNirmsPage) match {
+                case Right(true)  => Right(true)
+                case Right(false) =>
+                  Left(NonEmptyChain.one(UnexpectedPage(HasNirmsUpdatePage)))
+              }
+            } else {
+              Right(false)
+            }
+          case Left(_)            => Left(NonEmptyChain.one(UnexpectedPage(HasNirmsUpdatePage)))
```

Comment:
i would say this has to be page missing instead of unexpected page because if you start on the nirms journey hasNirms should always be answered but nirms number could be unexpected